src/components/AddControlKey.svelte

```diff
@@ -0,0 +1,123 @@
+<script lang="ts">
+    import {dotApi, storeConnected, storeCurrentAction, transactionSigningAddress} from "$lib/stores";
+    import type {ApiPromise} from "@polkadot/api";
+    import { submitAddControlKey } from "$lib/connections";
+    import {ActionForms, defaultDotApi} from "$lib/storeTypes";
+    import KeySelection from "./KeySelection.svelte";
+    import {onMount} from "svelte";
+    import {isFunction} from "@polkadot/util";
+    import { isLocalhost} from "$lib/utils";
+    import TransactionStatus from "./TransactionStatus.svelte";
+
+    let connected = false;
+    let thisDotApi = defaultDotApi;
+    let signingAddress: string = '';
+    let showSelf: boolean = false;
+    let selectedKey: string = '';
+    let web3FromSource;
+    let web3Enable;
+    let showTransactionStatus = false;
+    export let txnStatuses: Array<string> = []
+
+    onMount(async () => {
+        const extension = await import("@polkadot/extension-dapp");
+        web3FromSource = extension.web3FromSource;
+        web3Enable = extension.web3Enable;
+    })
+
+    export let providerId = 0;
+    export let validAccounts = {};
+
+    storeConnected.subscribe(val => connected = val);
+    dotApi.subscribe(api => thisDotApi = api);
+    transactionSigningAddress.subscribe(val => signingAddress = val);
+    storeCurrentAction.subscribe(val => showSelf = val == ActionForms.AddControlKey);
+
+    const hideSelf = () => {
+        storeCurrentAction.update(val => val = ActionForms.NoForm);
+        clearTxnStatuses();
+    }
+
+    const addNewTxnStatus = (txnStatus: string) => {
+        txnStatuses = [...txnStatuses, txnStatus];
+    }
+    const clearTxnStatuses = () => txnStatuses = new Array<string>();
+
+    const addControlKey = async (evt: Event) => {
+        clearTxnStatuses();
+        let endpointURI: string = thisDotApi.selectedEndpoint || '';
+        evt.preventDefault();
+        if (selectedKey === '') {
+            alert("Please choose a key to add.")
+        } else if (isFunction(web3FromSource) && isFunction(web3Enable)){
+            let newKeys = validAccounts[selectedKey];
+            let signingKeys = validAccounts[signingAddress]
+            showTransactionStatus = true;
+            if (isLocalhost(endpointURI)) {
+                await submitAddControlKey(
+                    thisDotApi.api as ApiPromise,
+                    undefined,
+                    newKeys,
+                    signingKeys,
+                    providerId,
+                    endpointURI as string,
+                    addNewTxnStatus,
+                );
+            } else {
+                const extensions = web3Enable('Frequency parachain provider dashboard: Adding Keys');
+                if (extensions.length !== 0) {
+                    const injectedExtension = await web3FromSource(signingKeys.meta.source);
+                    await submitAddControlKey(
+                        thisDotApi.api as ApiPromise,
```

Comment:
This is a lot of parameters and maybe when we do another extrinsic we can figure out a better interface.